lib/agent/socket_util.js

```diff
@@ -229,7 +229,11 @@ function validateCertChain(cert, cb)
       // if we have an ocsp response, cache it
       if (data && (!data.err || isValidOCSPError(data.err)))
       {
-        getOcspResponseCache().set(cert, data.res);
+        // check if cache is initialized before setting entry
+        if (getOcspResponseCache().isInitialized())
```

Comment:
1. If the cache hasn't been initialized we should initialize it as well and then set the cert. 
2. If we've already put this if condition do we still need the try catch block? I don't mind putting it just that I want to understand what is the actual cause of the exception so that we address that.

Reply:
The cause of this issue is that the driver is timing out trying to reach ocsp.snowflakecomputing.com which prevents the cache from being initialized. Do you still want to try initialize again here incase of a one off failure? We don't need the try catch block for this specific error if we have the condition. I kept it incase some other error happened.

Reply:
Yes we should initialize the cache nevertheless so the driver does not have to continuously reach out to the OCSP Responders again n again. I don't mind the try catch block but I'd recommend to add an ability to put the driver in test mode so that in test mode the driver does not suppress exceptions. But this is a general advise.